src/antlr/index.ts

```diff
@@ -20,8 +20,10 @@ class SqlParseTree {
     return this.tree.accept(visitor);
   }
 
-  getLineage<TableData extends { id: string }, ColumnData extends { id: string }>(
-    getTable: (id: TablePrimary) => { table: TableData; columns: ColumnData[] } | undefined
+  getLineage<TableData, ColumnData>(
+    getTable: (
+      id: TablePrimary
+    ) => { table: { id: string; data: TableData }; columns: { id: string; data: ColumnData }[] } | undefined
```

Comment:
Could be nice to factor out the getTable type into a reusable type definition that is also used in the LineageContext constructor. It would also make this function signature and the LineageContext constructor more readable.